pyincore/utils/analysisutil.py

```diff
@@ -68,35 +69,40 @@ def float_list_to_decimal(num_list: list):
     def float_dict_to_decimal(num_dict: dict):
         return {key: Decimal(str(num_dict[key])) for key in num_dict}
 
+    @staticmethod
+    def dmg_string_dict_to_dmg_float_dict(dmg_dict: dict):
```

Comment:
The logic looks good but I would give it more generic naming. 
The main function is to turn empty string to nan so maybe name it something to reflect that?  
Feel free to disregard my comment though it's not super necessary :-)